File: solutions/go/series/2/series.go
package series

func All(n int, s string) []string {
	res := make([]string, 0)
    if len(s) < n || n <= 0 {
        return res
    }
    if len(s) == n {
        res = append(res, s)
        return res
    }
    for i := 0; i < len(s) - n + 1; i++ {
        res = append(res, s[i:i + n])
    }
    return res
}

func UnsafeFirst(n int, s string) string {
    if len(s) <= n {
        return s
    }
    return s[0:n]
}

File: solutions/python/series/1/series.py
def slices(series, length):
    res = []
    s_len = len(series)
    if s_len == 0:
        raise ValueError("series cannot be empty")
    if s_len < length:
        raise ValueError("slice length cannot be greater than series length")
    if length == 0:
        raise ValueError("slice length cannot be zero")
    if length < 0:
        raise ValueError("slice length cannot be negative")
    if s_len == length:
        res.append(series)
        return res
    for index in range(0, s_len - length + 1):
        res.append(series[index:index + length])

    return res
